doc/ir-semantics.text

```diff
@@ -100,49 +110,62 @@ Update (partial) function, returns a S:
 
 ## Reduction relation '-->'
 
-  (P, T, H, E*, C* C) --> M
-    when M := step P(C) (P, T, H, E*, C* C)
+  (P, T, H, E*, π* π) --> M
+    when M := step P(π) (P, T, H, E*, π* π)
 
   step   CC[e]
-         (P, T, H, E*, C*)
-       = step CC[v] (P, T, H, E*, C*)
+         (P, T, H, E*, π*)
+       = step CC[v] (P, T, H, E*, π*)
     when v := eval P H E e
 
+  step   CC[call f (v, ...)]
+         (P, T, H, E*, π*)
+       = (P, T, H, E* E', π* π)
+    when P(f)  = Function (x, ...)
+     and E'   := (x ↦ v), ...
+     and π    := (f, active, start)
+
+  # Note: this works since syntactically every i has at most one call
+  step   (return v)
+         (P, T, H, E* E', π* π π')
+       = step cs (P, T, H, E*, π* π)
+    when cs := P(π)[call(...)/v]
+
   step   (print x)
-         (P, T, H, E*, C*)
-       = (P, (T, (H,E)[x]), H, E*, stepi C*)
+         (P, T, H, E*, π*)
+       = (P, (T, (H,E)[x]), H, E*, stepi π*)
 
   step   (branch v l₁ l₂)
-         (P, T, H, E*, C* (s, l))
-       = (P, T, H, E*, C* (s, l₁))
+         (P, T, H, E*, π* (f, s, l))
+       = (P, T, H, E*, π* (f, s, l₁))
     when v = true
 
   step   (branch v l₁ l₂)
-         (P, T, H, E*, C* (s, l))
-       = (P, T, H, E*, C* (s, l₂))
+         (P, T, H, E*, π* (f, s, l))
+       = (P, T, H, E*, π* (f, s, l₂))
     when v = false
 
   step   (const x = v)
-         (P, T, H, E* E, C*)
-       = (P, T, H, E* (E, x ↦ v), stepi C*)
+         (P, T, H, E* E, π*)
+       = (P, T, H, E* (E, x ↦ v), stepi π*)
 
   step   (mut x = v)
-         (P, T, H, E* E, C*)
-       = (P, T, H[a ↦ v], E* E[x ↦ a], stepi C*)
+         (P, T, H, E* E, π*)
+       = (P, T, H[a ↦ v], E* E[x ↦ a], stepi π*)
     when a fresh
 
   step   (x ← v)
-         (P, T, H, E* E, C*)
-       = (P, T, (H,E)[x ← v], E* E, stepi C*)
+         (P, T, H, E* E, π*)
+       = (P, T, (H,E)[x ← v], E* E, stepi π*)
 
   step   osr(v, _, _)
-         (P, T, H, E* E, C*)
-       = (P, T, H, E* E, stepi C*)
+         (P, T, H, E* E, π*)
+       = (P, T, H, E* E, stepi π*)
     when v = false
 
   step   osr(e, v', l', (x = ee, ...))
```

Comment:
Yeah, sorry, I rushed and also had a typo.

`e` should be `v` and `v` should be `s`. And then `v'` should be `s'`.